### PR TITLE
feat(insights): Insight editor panels

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Set out in https://github.com/PostHog/posthog/issues/8929

## Changes

- Adds a flag `8929-insight-editor-panels`, everything new is behind it.
- Implements a side-panel for all the insight types
- The Edit button now just shows/hides the panel.

## How did you test this code?

Methodically one by one, swapped the feature flag and checked that each "EditorFilter" element looks and behaves as before.

## Future Ideas
- [ ] Toggling edit could toggle the nav sidebar away temporarily to give more space
- [x] Animate in the editor panel

### CODE REVIEW

**Comment by benjackwhite** [comment]
This is a pretty large PR in the end and has a lot of copied files in order to allow the feature flag.

@Twixes what do you think to:
1. Merging as is
2. Refactoring quickly the old "copies" of the code to use the new `EF...` components so that at least any future changes to the subelements will be made in the same place?

File: frontend/src/lib/components/LemonCheckbox/LemonCheckbox.tsx [hunk 1]
```diff
@@ -75,7 +77,7 @@ export function LemonCheckbox({
                         checked={localChecked}
                         defaultChecked={defaultChecked}
                         onChange={(e) => {
-                            setLocalChecked(e.target.checked)
+                            checked === undefined && setLocalChecked(e.target.checked)
```
Comment (mariusandra) [inline frontend/src/lib/components/LemonCheckbox/LemonCheckbox.tsx:80]: Hmm... This is kind of hard to follow. I guess instead of here, `localChecked` will get set in `useEffect` for the case of a controlled input? Might be good to explicitly specify this with a comment here.

File: frontend/src/lib/components/PropertyGroupFilters/PropertyGroupFilters.tsx [hunk 2]
```diff
@@ -43,36 +44,29 @@ export function PropertyGroupFilters({
         duplicateFilterGroup,
     } = useActions(propertyGroupFilterLogic(logicProps))
 
-    // Update the logic's internal filters when the props change
-    useEffect(() => {
-        setFilters(propertyFilters ?? { type: FilterLogicalOperator.And, values: [] })
-    }, [propertyFilters])
```
Comment (mariusandra) [inline frontend/src/lib/components/PropertyGroupFilters/PropertyGroupFilters.tsx:49]: I'm not a fan of these useEffects, but I think it's needed. Otherwise the component will never update if its value prop (`propertyFilters`) changes.

File: frontend/src/scenes/insights/EditorFilters/EFFunnelsAdvanced.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                        <Tag className="lemonade-tag">{advancedOptionsUsedCount}</Tag>
+                    )}
+                </h4>
+                <div>
+                    <div className={clsx('advanced-options-dropdown', filters.funnel_advanced && 'expanded')}>
+                        <IconArrowDropDown />
+                    </div>
+                </div>
+            </div>
+            {filters.funnel_advanced ? (
```
Comment (mariusandra) [inline frontend/src/scenes/insights/EditorFilters/EFFunnelsAdvanced.tsx:37]: Somehow this never opens.

File: frontend/src/scenes/insights/EditorFilters/EFPathsAdvanced.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import { InfoCircleOutlined, SettingOutlined } from '@ant-design/icons'
+
+import { PayCard } from 'lib/components/PayCard/PayCard'
+import { preflightLogic } from 'scenes/PreflightCheck/preflightLogic'
+import { pathsLogic } from 'scenes/paths/pathsLogic'
+import { InputNumber, Tooltip } from 'antd'
+import { Link } from 'lib/components/Link'
+import { PathCleanFilterInput } from '../InsightTabs/PathTab/PathCleanFilterInput'
+
+export function EFPathsAdvanced({ insightProps }: EditorFilterProps): JSX.Element {
```
Comment (mariusandra) [inline frontend/src/scenes/insights/EditorFilters/EFPathsAdvanced.tsx:13]: This component seems kind of cramped, though happy to defer to another PR.
image

File: frontend/src/scenes/insights/EditorFilters/EditorFilterGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface EditorFilterGroupProps {
+    editorFilterGroup: InsightEditorFilterGroup
+    insight: Partial<InsightModel>
+    insightProps: InsightLogicProps
+}
+
+export function EditorFilterGroup({ editorFilterGroup, insight, insightProps }: EditorFilterGroupProps): JSX.Element {
+    const { title, editorFilters, defaultExpanded } = editorFilterGroup
+    const [isRowExpanded, setIsRowExpanded] = useState(defaultExpanded)
```
Comment (mariusandra) [inline frontend/src/scenes/insights/EditorFilters/EditorFilterGroup.tsx:17]: Scope creep alert (please defer): I wonder if we should also store the expanded/collapsed state in a filter, and save it with the insight? Perhaps not?

File: frontend/src/scenes/insights/EditorFilters/EditorFilterGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                    fullWidth
+                    onClick={() => setIsRowExpanded(!isRowExpanded)}
+                    sideIcon={isRowExpanded ? <IconUnfoldLess /> : <IconUnfoldMore />}
+                    title={isRowExpanded ? 'Show less' : 'Show more'}
+                    style={{
+                        fontWeight: 600,
+                    }}
+                >
+                    {title}
+                </LemonButton>
```
Comment (mariusandra) [inline frontend/src/scenes/insights/EditorFilters/EditorFilterGroup.tsx:32]: We could use a `data-attr` on these.

**Review by mariusandra — COMMENTED** [review]
Looking good! Found a few things to comment on before the green checkmark though :)

File: frontend/src/lib/components/LemonCheckbox/LemonCheckbox.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/lib/components/LemonCheckbox/LemonCheckbox.tsx:80]: Yeah exactly. Without this it creates a double-click effect. I'll add a comment

File: frontend/src/scenes/insights/EditorFilters/EFFunnelsAdvanced.tsx (hunk 3 above)
Comment (benjackwhite) [reply frontend/src/scenes/insights/EditorFilters/EFFunnelsAdvanced.tsx]: This is fixed now - removed all of that checking as it is basically handled by the parent component now

File: frontend/src/scenes/insights/EditorFilters/EditorFilterGroup.tsx (hunk 5 above)
Comment (benjackwhite) [reply frontend/src/scenes/insights/EditorFilters/EditorFilterGroup.tsx:17]: I thought about this too but I think the potential branches are weird. I would vote we play with it and see what feels rights, gather some data and make a decision then

File: frontend/src/scenes/insights/EditorFilters/EditorFilterGroup.tsx (hunk 6 above)
Comment (benjackwhite) [reply frontend/src/scenes/insights/EditorFilters/EditorFilterGroup.tsx]: Added 👍

**Review by benjackwhite — APPROVED** [review]
Some of the best code I have ever seen. Great work 😉
